include/fmt/format-inl.h

```diff
@@ -29,6 +29,10 @@
 #    include <windows.h>
 #    undef WIN32_LEAN_AND_MEAN
 #    undef NOMINMAX
+#  elif defined(NOMINMAX) && !defined(WIN32_LEAN_AND_MEAN) 
+#    define WIN32_LEAN_AND_MEAN 
+#    include <windows.h> 
+#    undef WIN32_LEAN_AND_MEAN 
 #  else
 #    include <windows.h>
 #  endif
```

Comment:
I think this whole block should be replaced with something like

```c++
#  ifdef FMT_BIG_WIN
#    include <windows.h>
#  else
#    define NOMINMAX
#    define WIN32_LEAN_AND_MEAN
#    include <windows.h>
#    undef WIN32_LEAN_AND_MEAN
#    undef NOMINMAX
#  endif
```
Then the default will be less broken windows headers and whoever wants all the garbage can define `FMT_BIG_WIN`.